File: src/main/resources/output/IFC4X3_js/Entities/IfcRoot/IfcRelationship/IfcRelConnects/IfcRelInterferesElements/IfcRelInterferesElements.js
class IfcRelInterferesElements extends IfcRelConnects {
    constructor() {
        /** @type {IFCINTERFERENCESELECT} */
        this.RelatingElement = null;
        /** @type {IFCINTERFERENCESELECT} */
        this.RelatedElement = null;
        /** @type {IFCCONNECTIONGEOMETRY} */
        this.InterferenceGeometry = null;
        /** @type {IFCIDENTIFIER} */
        this.InterferenceType = null;
        /** @type {IFCLOGICAL} */
        this.ImpliedOrder = null;
        /** @type {IFCSPATIALZONE} */
        this.InterferenceSpace = null;
    }

    // === WHERE CLAUSES ===
    // NoSelfReference : RelatingElement :<>: RelatedElement
}

File: src/main/resources/output/IFC4X3_java/Entities/IfcRoot/IfcRelationship/IfcRelConnects/IfcRelInterferesElements/IfcRelInterferesElements.java
public class IfcRelInterferesElements extends IfcRelConnects {
    public IfcInterferenceSelect RelatingElement;
    public IfcInterferenceSelect RelatedElement;
    public IfcConnectionGeometry InterferenceGeometry;
    public IfcIdentifier InterferenceType;
    public IfcLogical ImpliedOrder;
    public IfcSpatialZone InterferenceSpace;

    // === WHERE CLAUSES ===
    // NoSelfReference : RelatingElement :<>: RelatedElement
}
